src/Api/Projects.php

```diff
@@ -1035,13 +1035,17 @@ public function languages($project_id)
 
     /**
      * @param int|string $project_id
-     * @param array      $parameters
+     * @param array      $parameters {
+     *
+     *     @var string $search         Return list of forks matching the search criteria (optional)
+     * }
      *
      * @return mixed
      */
     public function forks($project_id, array $parameters = [])
     {
         $resolver = $this->createOptionsResolver();
+        $resolver->setDefined('search');
```

Comment:
Shouldn't this be defined in the options resolver function?

Reply:
I'm not too familiar with the code inside the library, but the rest of the methods seem to be defining the options in the same way, within the method.

Reply:
I've done it now based on the current code (I took the `all` method as reference) and the information found here: https://docs.gitlab.com/ee/api/projects.html#list-forks-of-a-project

Reply:
I added all the other options available.

Reply:
Yeh, this is not correct. Please look over the `createOptionsResolver` method in this class, and only add the things not already in there, or if search is always applicable, instead, add it there.

Reply:
I don't think that `search` should bubble up to the `createOptionsResolver` as that's used by many other methods. Note that there is no override for `createOptionsResolver` in this class, it is using the one of `AbstractApi.php` file.

The line `$resolver->setDefined('search');` was taken from the `all` method within this same class.

Not sure what more changes are required? I might need some more details.